example/scalalib/native/2-interop/build.mill

```diff
@@ -0,0 +1,66 @@
+package build
+import mill._, scalalib._, scalanativelib._
+
+object `package` extends RootModule with ScalaNativeModule {
+  def scalaVersion = "2.13.11"
+  def scalaNativeVersion = "0.5.5"
+
+  object test extends ScalaNativeTests {
+    def ivyDeps = Agg(ivy"com.lihaoyi::utest::0.8.4")
+    def testFramework = "utest.runner.Framework"
+  }
+  
+  def nativeLinkingOptions = Seq("-L" + millSourcePath.toString + "/target")
+
+  // Compiled C
+  def nativeCompiled = T {
+    os.makeDir.all(millSourcePath / "target")
+
+    os.proc("gcc", "-m64", "-shared",
+      "-c", millSourcePath.toString + "/resources/scala-native/HelloWorld.c",
+      "-o", millSourcePath.toString + "/target/libHelloWorld.so"
```

Comment:
Keeping `.c` files outside `resources/scala-native` directory cause errors. Why do you want it to be in `src-native` and not `resources/scala-native`?

Reply:
Because resources are meant to be served at runtime, and c sources are not that. Causing errors isnt a problem, you just need to investigate and fix them

Reply:
I was thingking scala native code expected c file to be in `resources/scala-native`
https://scala-native.org/en/stable/user/native.html#using-libraries-with-native-code

Reply:
Can you check to see if the c sources are relocatable? I would assume its configurable by a flag or something, but might need to ask the scala native folks if they know

Reply:
If it's not relocatable, then we can go with as is, but we should dig into it a bit to make sure

Reply:
Okay... I need tried but got this error

```
build.mill-57] [info] Linking (multithreadingEnabled=true, disable if not used) (1974 ms)
[build.mill-57] [info] Discovered 869 classes and 5463 methods after classloading
[build.mill-57] [info] Checking intermediate code (quick) (95 ms)
[build.mill-57] [info] Multithreading was not explicitly enabled - initial class loading has not detected any usage of system threads. Multithreading support will be disabled to improve performance.
[build.mill-57] [info] Linking (multithreadingEnabled=false) (750 ms)
[build.mill-57] [info] Discovered 478 classes and 2491 methods after classloading
[build.mill-57] [info] Checking intermediate code (quick) (8 ms)
[build.mill-57] [info] Discovered 461 classes and 1881 methods after optimization
[build.mill-57] [info] Optimizing (debug mode) (2119 ms)
[build.mill-57] [info] Produced 3 LLVM IR files
[build.mill-57] [info] Generating intermediate code (2836 ms)
[build.mill-57] [info] Compiling to native code (5318 ms)
[build.mill-57] [info] Linking with [pthread, dl]
[build.mill-57] [error] /usr/bin/ld: /workspaces/test/saturday/mill/example/scalalib/native/2-interop/out/nativeWorkdir.dest/native/generated/1.ll.o: in function `_SM9foo.Main$D4mainLAL16java.lang.String_uEO':
[build.mill-57] [error] 1.ll:(.text+0x26f): undefined reference to `reverseString'
[build.mill-57] [error] clang: error: linker command failed with exit code 1 (use -v to see invocation)
[build.mill-57] [info] Total (12026 ms)
```

The file tree
 ```
 tree
.
├── build.mill
├── resources
├── scala-native
│   └── HelloWorld.c
├── src
│   └── foo
│       └── HelloWorld.scala
└── test
    └── src
        └── HelloWorldTests.scala

6 directories, 4 files
```

Online changes I made in build.mill is this
```
  // Additional source folder to put C sources
  def nativeSources = Task.Sources(millSourcePath / "scala-native")
```

I tried with the `2-interop` example

Reply:
but when I structure the project to use the `resources/scala-native` and  revert back the changes made in `build.mill` it works

Reply:
I just remeber please check this link to the scala forum where I asked a question about an error last week or so
https://users.scala-lang.org/t/how-to-test-scala-native-code-interop-with-c/10314/3?u=c0d33ngr

Reply:
@c0d33ngr ok, lets leave the code as is then. But lets add a comment linking to the documentstion and linking to the stackoverflow thread, so the reasoning is clear for anyone else who reads the code in future and wonders why we are doing that

Reply:
Okay. I'll do for both `2-interop` and `3-multi-module` examples